Tests/SwiftDocCTests/Utility/DataStructures/GroupedSequenceTests.swift

```diff
@@ -0,0 +1,48 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import XCTest
+@testable import SwiftDocC
+
+class GroupedSequenceTests: XCTestCase {
+    /// A grouped sequence of strings, with their number of characters as the key.
+    var groupedSequence = GroupedSequence<Int, String>(deriveKey: \.count)
+    
+    func testAppends() {
+        groupedSequence.append("a")
+        groupedSequence.append("aa")
+        
+        XCTAssertEqual(groupedSequence[1], "a")
+        XCTAssertEqual(groupedSequence[2], "aa")
```

Comment:
you probably want to add the following to this:
```
groupedSequence.append("b")
XCTAssertEqual(groupedSequence[1], "b")
```

Reply:
Yes, that's a great idea!